src/recipe_generator/pypi.rs

```diff
@@ -232,10 +232,10 @@ pub async fn generate_pypi_recipe(opts: &PyPIOpts) -> miette::Result<()> {
     // replace URL with the shorter version that does not contain the hash
     let release_url = if release.url.starts_with("https://files.pythonhosted.org/") {
         let simple_url = format!(
-            "https://pypi.io/packages/source/{}/{}/{}-{}.tar.gz",
+            "https://pypi.org/packages/source/{}/{}/{}-{}.tar.gz",
             &info.name.to_lowercase()[..1],
             info.name.to_lowercase(),
-            info.name.to_lowercase(),
+            info.name.to_lowercase().replace("-", "_"),
```

Comment:
I think this is great! Although the actual rules are a bit more involved: https://github.com/prefix-dev/rattler-build/issues/1363#issuecomment-2607786942

The way I understand it:

- any sequence of `[-.` is going to be replaced by a _single_ `_`.